server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -501,31 +502,16 @@ protected void onShardResult(Result result) {
             logger.trace("got first-phase result from {}", result != null ? result.getSearchShardTarget() : null);
         }
         // clean a previous error on this shard group (note, this code will be serialized on the same shardIndex value level
-        // so its ok concurrency wise to miss potentially the shard failures being created because of another failure
+        // so it's ok concurrency wise to miss potentially the shard failures being created because of another failure
         // in the #addShardFailure, because by definition, it will happen on *another* shardIndex
         AtomicArray<ShardSearchFailure> shardFailures = this.shardFailures.get();
         if (shardFailures != null) {
             shardFailures.set(result.getShardIndex(), null);
         }
-        results.consumeResult(result, this::onShardResultConsumed);
-    }
-
-    private void onShardResultConsumed() {
-        successfulOps.incrementAndGet();
-        // we need to increment successful ops first before we compare the exit condition otherwise if we
```

Comment:
This comment makes no sense at this point, there is no total ops counter anymore.